samples/test-app/src/androidTest/kotlin/com/avito/android/ui/test/SimpleRuleTest.kt

```diff
@@ -0,0 +1,52 @@
+package com.avito.android.ui.test
+
+import com.avito.android.rule.SimpleRule
+import org.hamcrest.Matchers.isA
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+
+class SimpleRuleTest {
+
+    @get:Rule
+    val rule = FailedRule()
+
+    @get:Rule
+    var thrown: ExpectedException = ExpectedException.none()
+
+    @Test
+    fun has_test_error__test_rule_failed_after_test_execution() {
+        rule.failAfterTest = true
+
+        thrown.expect(RuleError::class.java)
+        thrown.expectCause(isA(IllegalStateException::class.java))
+
+        throw IllegalStateException("Error from test")
+    }
+
+    @Test
+    fun has_test_error__test_failed() {
+        rule.failAfterTest = false
+
+        thrown.expect(IllegalStateException::class.java)
+        thrown.expectMessage("Error from test")
+
+        throw IllegalStateException("Error from test")
+    }
+
+    private class RuleError : java.lang.RuntimeException()
+
+    class FailedRule : SimpleRule() {
```

Comment:
Technically, I can test changes in unit tests. I wanted to cover them in integration inside AndroidJUnitRunner to be sure.